### PR TITLE
[SPARK-29305][BUILD] Update LICENSE and NOTICE for Hadoop 3.2

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
This PR update LICENSE and NOTICE for Hadoop 3.2. Hadoop 3.2 newly added jars:

```
com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.9.5	
com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.5	
com.fasterxml.woodstox:woodstox-core:5.0.3	
com.github.stephenc.jcip:jcip-annotations:1.0-1	
com.google.re2j:re2j:1.1		
com.microsoft.sqlserver:mssql-jdbc:6.2.1.jre7	
com.nimbusds:nimbus-jose-jwt:4.41.1		
dnsjava:dnsjava:2.1.7		
net.minidev:accessors-smart:1.2	
net.minidev:json-smart:2.3	
org.apache.commons:commons-configuration2:2.1.1	
org.apache.geronimo.specs:geronimo-jcache_1.0_spec:1.0-alpha-1	
org.apache.hadoop:hadoop-hdfs-client:3.2.0 
org.apache.kerby:kerb-admin:1.0.1	
org.apache.kerby:kerb-client:1.0.1	
org.apache.kerby:kerb-common:1.0.1	
org.apache.kerby:kerb-core:1.0.1	
org.apache.kerby:kerb-crypto:1.0.1	
org.apache.kerby:kerb-identity:1.0.1	
org.apache.kerby:kerb-server:1.0.1	
org.apache.kerby:kerb-simplekdc:1.0.1	
org.apache.kerby:kerb-util:1.0.1	
org.apache.kerby:kerby-asn1:1.0.1	
org.apache.kerby:kerby-config:1.0.1	
org.apache.kerby:kerby-pkix:1.0.1	
org.apache.kerby:kerby-util:1.0.1	
org.apache.kerby:kerby-xdr:1.0.1	
org.apache.kerby:token-provider:1.0.1	
org.codehaus.woodstox:stax2-api:3.1.4	
org.ehcache:ehcache:3.3.1		
```
### Why are the changes needed?
We will distribute a binary release based on Hadoop 3.2 / Hive 2.3 in future.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
N/A

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
@wangyum @srowen @dongjoon-hyun

**Comment by wangyum** [comment]
ok to test

**Review by wangyum — COMMENTED** [review]
Please update PR title to `[SPARK-29305]...`.

File: LICENSE-binary [hunk 1]
```diff
@@ -218,6 +218,7 @@ javax.jdo:jdo-api
 joda-time:joda-time
 net.sf.opencsv:opencsv
 org.apache.derby:derby
+org.ehcache:ehcache
```
Comment (srowen) [inline LICENSE-binary]: ehcache is ALv2, good; I think there's a NOTICE entry for it though:
https://github.com/ehcache/ehcache3/blob/master/NOTICE
(Not sure what version this is pulling in; this is 3.x)

File: LICENSE-binary [hunk 2]
```diff
@@ -292,14 +310,20 @@ com.fasterxml.jackson.core:jackson-annotations
...
 com.fasterxml.jackson.module:jackson-module-paranamer
 com.fasterxml.jackson.module:jackson-module-scala_2.12
+com.fasterxml.woodstox:woodstox-core
 com.github.mifmif:generex
+com.github.stephenc.jcip:jcip-annotations
 com.google.code.findbugs:jsr305
 com.google.code.gson:gson
 com.google.inject:guice
 com.google.inject.extensions:guice-servlet
+com.google.re2j:re2j
```
Comment (srowen) [inline LICENSE-binary]: This looks like 3-clause BSD, not ALv2, so it should be later in the file:
https://github.com/google/re2j/blob/master/LICENSE

We should also copy its license into licenses-binary

File: LICENSE-binary [hunk 3]
```diff
@@ -401,13 +428,15 @@ BSD 2-Clause
 ------------
 
 com.github.luben:zstd-jni
+dnsjava:dnsjava
```
Comment (srowen) [inline LICENSE-binary]: Likewise add copies of the licenses below to licenses-binary/

**Review by srowen — CHANGES_REQUESTED** [review]

**Comment by AngersZhuuuu** [comment]
> Please update PR title to `[SPARK-29305]...`.

Sorry for my mistake when create new branch..

File: LICENSE-binary (hunk 1 above)
Comment (AngersZhuuuu) [reply LICENSE-binary]: Version is right.  3.3.1.  The NOTICE file is not under META-INF, miss this file, thanks a lot.

File: LICENSE-binary (hunk 3 above)
Comment (AngersZhuuuu) [reply LICENSE-binary]: > Likewise add copies of the licenses below to licenses-binary/

Done.

File: LICENSE-binary [hunk 4]
```diff
@@ -541,3 +571,67 @@ data/mllib/images/kittens/54893.jpg
 data/mllib/images/kittens/DP153539.jpg
 data/mllib/images/kittens/DP802813.jpg
 data/mllib/images/multi-channel/chr30.4.184.jpg
+
+
+Golang LICENSE derived from Russ Cox's RE2
```
Comment (srowen) [inline LICENSE-binary]: Sorry, I mean this needs to be a text file under licenses-binary/

**Review by srowen — CHANGES_REQUESTED** [review]

File: LICENSE-binary (hunk 4 above)
Comment (AngersZhuuuu) [reply LICENSE-binary]: > Sorry, I mean this needs to be a text file under licenses-binary/

Add it, the file name is ok? for `re2j`, it is personal license follow go license. and for `dnsjava` also personal license.

**Review by srowen — APPROVED** [review]
That's fine. The licenses are standard, not 'personal'. Authors retain copyright and can say so; it doesn't affect the license.

**Comment by AngersZhuuuu** [comment]
> That's fine. The licenses are standard, not 'personal'. Authors retain copyright and can say so; it doesn't affect the license.

I see, thanks a lot for your guidance.

**Pull request closed without merging**